### ISSUE TITLE
Make the mobile phone preview resizable.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#1721]
Currently our mobile mode preview phone is in landscape mode and only has one size. We can add a handle on the bottom right that allows people to resize it as they wish, like seen in this screenshot from Chrome's dev tools:

![resize-mobile](https://cloud.githubusercontent.com/assets/25212/22807798/aa33f572-eedd-11e6-8986-a4c25c55d67f.gif)

In addition, we'd need to remember this setting so when the editor refreshes, we persist this size.

**Comment by humphd** [comment]
I'm not sure if this makes sense in the context of it looking like a phone.  We aren't giving a true responsive view, we're giving a single width/height viewport that simulates a phone.  If we switch to what you have here, we might need to rethink the phone wrapper.

**Comment by flukeout** [comment]
Yeah definitely - I think having the flexible sizing outweighs the current aesthetics, as amazing as they are! Another option would be to add a 'rotate' button to the phone to simulate landscape mode. Or, maybe another click on the phone preview icon could do that.

Let's dive in further if someone picks this up.

**Comment by rashmimalpande** [comment]
Hi, I would like to try this.  Are we talking about something like this -  

![moz1](https://cloud.githubusercontent.com/assets/14840334/23022744/760d10e4-f478-11e6-8455-6eea1b15587a.jpg)

I just tried adding icons through dev tools.

**Comment by humphd** [comment]
@flukeout can respond to this with thoughts.

**Comment by gideonthomas** [comment]
@flukeout just a suggestion: I think we might want to start of with just providing a few default sizes (with rotation I guess) similar to what we get from devtools rather than going for full responsiveness atm.

**Comment by flukeout** [comment]
Hey @rashmimalpande - thanks for getting this started! I think most useful thing we can offer is to allow people to switch the mobile preview to landscape mode. They can already resize it pretty easily by changing the width of the preview so I don't think we need to worry about different types of mobile views.

I think we can put a control to change the orientation somewhere in the area around the mobile phone preview itself and not in the tool bar, something like this...

<img width="727" alt="image" src="https://cloud.githubusercontent.com/assets/25212/23086463/d6fe9dba-f521-11e6-81d5-bb6088280bde.png">

Anyway, let's give that a try and see how it feels. I can provide guidance on styling & the icon.

Also, if you're feeling ambitious, you could try to make the phone resizable with jQuery resizable

* https://jqueryui.com/resizable/

This makes it pretty easy to change the size of elements, so might work well for us.

**Comment by rashmimalpande** [comment]
I will start working on this. Can you give me some idea about how should I start with this?

**Comment by humphd** [comment]
@rashmimalpande here are the files you'll care about (they all live in the mozilla/brackets repo):

* HTML for the mobile view: https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/lib/Mobile.html
* JS that controls the mobile view: https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/lib/UI.js#L245
* CSS for mobile view: starts here https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/stylesheets/style.css#L46

**Comment by rashmimalpande** [comment]
Thanks @humphd . I will start working on this.

**Comment by humphd** [comment]
@rashmimalpande how is this coming?

**Comment by rashmimalpande** [comment]
@humphd I am sorry. I didn't get much time to work on it. I have set up everything. I will work on it this week.

**Comment by humphd** [comment]
Moving this out of the Assigned label.  If someone wants to work on it, let us know.
